server/runtime/src/test/resources/configuration/ServerConfigurationParserTest.json

```diff
@@ -0,0 +1,440 @@
+{
```

Comment:
It's not possible to start the server if the `cache-container ` element is present. The default xml fails with:

```java
2021-08-09 11:24:43,762 FATAL (main) [org.infinispan.SERVER] ISPN080028: Infinispan Server failed to start org.infinispan.commons.CacheConfigurationException: ISPN000327: Cannot find a parser for element 'cache-container' in namespace 'urn:infinispan:config:13.0'. Check that your configuration is up-to date for Infinispan '13.0.0-SNAPSHOT' and if you have the proper dependency in the classpath
	at org.infinispan.configuration.parsing.ParserRegistry.findNamespaceParser(ParserRegistry.java:230)
	at org.infinispan.configuration.parsing.ParserRegistry.parseElement(ParserRegistry.java:203)
	at org.infinispan.configuration.parsing.ParserRegistry.parse(ParserRegistry.java:184)
	at org.infinispan.configuration.parsing.ParserRegistry.parse(ParserRegistry.java:170)
	at org.infinispan.configuration.parsing.ParserRegistry.parse(ParserRegistry.java:164)
	at org.infinispan.server.Server.parseConfiguration(Server.java:292)
	at org.infinispan.server.Server.<init>(Server.java:216)
	at org.infinispan.server.Bootstrap.runInternal(Bootstrap.java:158)
	at org.infinispan.server.tool.Main.run(Main.java:98)
	at org.infinispan.server.Bootstrap.main(Bootstrap.java:47)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.infinispan.server.loader.Loader.run(Loader.java:106)
	at org.infinispan.server.loader.Loader.main(Loader.java:51)
```

```yaml
{
   "cache-container": {
      "name": "default",
      "statistics": "true",
      "transport": {
         "cluster": "${infinispan.cluster.name:cluster}",
         "stack": "${infinispan.cluster.stack:tcp}",
         "node-name": "${infinispan.node.name:}"
      },
      "security": {
         "authorization": null
      }
   },
   "server": {
      "interfaces": {
         "interface": {
            "name": "public",
            "inet-address": {
               "value": "${infinispan.bind.address:127.0.0.1}"
            }
         }
      },
      "socket-bindings": {
         "default-interface": "public",
         "port-offset": "${infinispan.socket.binding.port-offset:0}",
         "socket-binding": [
            {
               "name": "default",
               "port": "${infinispan.bind.port:11222}"
            },
            {
               "name": "memcached",
               "port": "11221"
            }
         ]
      },
      "security": {
         "credential-stores": {
            "credential-store": {
               "name": "credentials",
               "path": "credentials.pfx",
               "clear-text-credential": {
                  "clear-text": "secret"
               }
            }
         },
         "security-realms": {
            "security-realm": {
               "name": "default",
               "properties-realm": {
                  "groups-attribute": "Roles",
                  "user-properties": {
                     "path": "users.properties"
                  },
                  "group-properties": {
                     "path": "groups.properties"
                  }
               }
            }
         }
      },
      "endpoints": {
         "socket-binding": "default",
         "security-realm": "default"
      }
   }
}
```

Removing the "cache-container" element allows the server to start. I tried adding `"_namespace": "urn:infinispan:config:13.0",` however this had no effect.

Reply:
There's an opportunity here to fix some arrays, looking by this JSON. Eg.:
```json
"interfaces": {
         "interface": {
            "name": "public",
            "inet-address": {
               "value": "${infinispan.bind.address:127.0.0.1}"
            }
         }
      }
```

would be better be represented as:

```json

    "interfaces":[
        {
            "name":"public",
            "inet-address":{
                "value":"${infinispan.bind.address:127.0.0.1}"
            }
        },
        {
            "name":"another",
            "inet-address":{
                "value":"${infinispan.bind.address:127.0.0.1}"
            }
        }
    ]

```

I assume there could be multiple interfaces? So it should be an array, like ```socket-binding```. The same applies to ```credential-stores```, ```security-realms``` and other maybe?

Reply:
That's because `cache-container` is not a top-level element in the parsers (`server` is).
The following will work:
```json
{
  "infinispan": {
    "cache-container": {
    },
    "server": {
    }
}
```

Reply:
:+1: That fixed it

Reply:
what about the arrays @tristantarrant

Reply:
If you look at the actual test file and ignore @ryanemerson 's json, you'll see that we are indeed using arrays for `interfaces` etc

Reply:
Thanks, I was wondering if that JSON was actually from this PR :)

Reply:
Sorry for the confusion. I just converted our `infinispan.xml` to json using the first converter I found.